apps/include/apps.h

```diff
@@ -102,7 +102,7 @@ typedef struct args_st {
 int wrap_password_callback(char *buf, int bufsiz, int verify, void *cb_data);
 
 int chopup_args(ARGS *arg, char *buf);
-# ifdef HEADER_X509_H
+# ifdef OPENSSL_X509_H
```

Comment:
> Supporting stupidity like that goes too far IMHO.

@bernd-edlinger it looks like your harsh https://github.com/openssl/openssl/pull/9274#issuecomment-507047651 also applies to our own code ;-)

Reply:
still does not look like a smart idea ;-)

Reply:
I removed the inclusion detectors for `HEADER_X509_H` entirely. As suspected, they are not necessary, because the X509 forward declarations have already been parsed.

There was another inclusion detector `HEADER_SSL_H` in s_apps.h.  I removed it and replaced it by an `#include <openssl/ssl.h>` after ensuring that `s_apps.h` is only included in modules which actually need it. (s_client.c, s_server.c, etc.)